ocaml/typing/typecore.ml

```diff
@@ -7995,7 +8023,10 @@ let report_error ~loc env = function
   | Apply_wrong_label (l, ty, extra_info) ->
       let print_label ppf = function
         | Nolabel -> fprintf ppf "without label"
-        | l -> fprintf ppf "with label %s" (prefixed_label_name l)
+        | Position _ as l -> fprintf ppf "with label %s:[%%src_pos]"
```

Comment:
I think this case can never happen, because we never find `Position` arguments in a list of arguments. I think it would be cleaner to `assert false` here and have a comment explaining why this is impossible.

Reply:
Note: comment somewhere that in applications, Positions will never be spotted